pkg/cloudencrypt/dual_test.go

```diff
@@ -0,0 +1,39 @@
+package cloudencrypt
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_DualEncryptor(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	secretKey, err := generateSecretKey()
+	assert.NoError(t, err)
+
+	nativeEncryptor, err := NewNativeEncryptor(secretKey)
+	assert.NoError(t, err)
+
+	encryptor, err := NewDualEncryptor(ctx, nativeEncryptor)
+	assert.NoError(t, err)
+
+	meta := MetadataKV{
+		Key:   "metakey",
+		Value: "metavalue",
+	}
+
+	ciphertext, err := encryptor.Encrypt(ctx, []byte("Lorem ipsum"), meta)
+	assert.NoError(t, err)
+
+	_, err = encryptor.Decrypt(ctx, ciphertext)
+	assert.ErrorContains(t, err, "cipher: message authentication failed")
```

Comment:
It would be nice to spot which branch returned this. This test tests only 1 case, but It seems like it does not test whole flow.
I would expect that dual encryptor uses NativeEncryptor + some of CloudEncryptor (e.g AWS)

Reply:
That's how it's used in practice. It always uses the NativeEncryptor internally to encrypt the given plaintext and then use the given encryptor to encrypt the secret key. The given encryptor will be gcp/aws/azure in practice but this is just a unit test so native works completely fine for this purpose.

Reply:
I understand, I will leave it but check code coverage please. According to my flow it sais the workflow is not covered correctly. 
```
go tool cover -html=/tmp/profile.out
```
I checked only files that I am able to test without specific cloud encryptor/decryptor

Reply:
If you mean the fact that the test doesn't cover the error paths, it's because there is no way to actually trigger those error cases in practice.